crates/release_plz_core/src/release.rs

```diff
@@ -184,15 +188,25 @@ async fn publish_release(
     repo: Repo,
     release_body: &str,
     git_token: SecretString,
+    backend: &GitBackend,
 ) -> anyhow::Result<()> {
     let repo_url = match repo_url {
         Some(url) => RepoUrl::new(url),
         None => RepoUrl::from_repo(&repo),
     }?;
-    if repo_url.is_on_github() {
-        let github = GitHub::new(repo_url.clone().owner, repo_url.clone().name, git_token);
-        let github_client = GitHubClient::new(&github)?;
-        let _page = github_client.create_release(&git_tag, release_body).await?;
+    match backend {
+        GitBackend::Github(_) => {
```

Comment:
```suggestion
        GitBackend::Github(github) => {
```
Here you already have the GitHub object, you don't need to rebuild it.
Same for gitea on line 203.